vector/src/main/java/im/vector/adapters/VectorMessagesAdapter.java

```diff
@@ -2572,27 +2579,31 @@ private void onMessageClick(final Event event, final String textMsg, final View
                 }
 
                 // offer to report a message content
-                menu.findItem(R.id.ic_action_vector_report).setVisible(!mIsPreviewMode && !TextUtils.equals(event.sender, mSession.getMyUserId()));
+                if (mContext.getResources().getBoolean(R.bool.hide_message_report_content)) {
+                    menu.findItem(R.id.ic_action_vector_report).setVisible(false);
+                } else {
+                    menu.findItem(R.id.ic_action_vector_report).setVisible(!mIsPreviewMode && !TextUtils.equals(event.sender, mSession.getMyUserId()));
+                }
             }
         }
 
         // e2e
-        menu.findItem(R.id.ic_action_device_verification).setVisible(mE2eIconByEventId.containsKey(event.eventId));
-
+        if (mContext.getResources().getBoolean(R.bool.hide_message_view_session)) {
+            menu.findItem(R.id.ic_action_device_verification).setVisible(false);
+        } else {
+            menu.findItem(R.id.ic_action_device_verification).setVisible(mE2eIconByEventId.containsKey(event.eventId));
+        }
         // display the menu
-        popup.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-            @Override
-            public boolean onMenuItemClick(final MenuItem item) {
-                // warn the listener
-                if (null != mVectorMessagesAdapterEventsListener) {
-                    mVectorMessagesAdapterEventsListener.onEventAction(event, textMsg, item.getItemId());
-                }
+        popup.setOnMenuItemClickListener(item -> {
```

Comment:
I'd suggest we refrain from making these code refactors, unless the same refactor is made to riot master, as it may cause merge conflict issues when we pull from riot master